pkgs/development/tools/watchman/default.nix

```
@@ -1,47 +1,90 @@
-{ stdenv, lib, config, fetchFromGitHub, autoconf, automake, pcre
-, libtool, pkg-config, openssl
+{ stdenv, lib, config, fetchFromGitHub, pcre, cmake, folly, fizz, libsodium, fb303, cpptoml, rustPlatform, libunwind, double-conversion, lz4, zstd
+, pkg-config, openssl, python3, gtest, glog, boost, libevent, fmt_9, zlib, wangle, fbthrift, edencommon, ensureNewerSourcesHook, libiconv
 , confFile ? config.watchman.confFile or null
-, withApple ? stdenv.isDarwin, CoreServices
+, CoreServices
 }:
 
 stdenv.mkDerivation rec {
   pname = "watchman";
-  version = "4.9.0";
+  version = "2023.01.16.00";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "watchman";
     rev = "v${version}";
-    sha256 = "0fdaj5pmicm6j17d5q7px800m5rmam1a400x3hv1iiifnmhgnkal";
+    sha256 = "sha256-r1j3TFTUFXncZSNdAwzUdqQb2QjCiXXclQ8DIZZ4lCU=";
   };
 
-  nativeBuildInputs = [ autoconf automake pkg-config libtool ];
-  buildInputs = [ pcre openssl ]
-    ++ lib.optionals withApple [ CoreServices ];
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=ON"
+    "-DENABLE_EDEN_SUPPORT=NO"
+    "-DWATCHMAN_STATE_DIR=/run/watchman"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "-DCMAKE_OSX_DEPLOYMENT_TARGET=10.14" # For aligned allocation
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    (ensureNewerSourcesHook {
+      year = "1980";
+    })
+  ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
+  buildInputs = [
+    pcre
+    openssl
+    python3
+    gtest
+    glog
+    boost
+    libevent
+    fmt_9
+    libsodium
+    zlib
+    folly
+    fizz
+    wangle
+    fbthrift
+    fb303
+    cpptoml
+    edencommon
+    libunwind
+    double-conversion
+    lz4
+    zstd
+    libiconv
+  ] ++ lib.optionals stdenv.isDarwin [ CoreServices ];
 
   configureFlags = [
     "--enable-lenient"
     "--enable-conffile=${if confFile == null then "no" else confFile}"
     "--with-pcre=yes"
-
-    # For security considerations re: --disable-statedir, see:
-    # https://github.com/facebook/watchman/issues/178
-    "--disable-statedir"
   ];
 
   prePatch = ''
```

```suggestion
  postPatch = ''
```
Also can we reduce the scope of selected files easily?